File: bantuin/amin/1.cpp
#include <iostream>
#include <cmath>

using namespace std;

double f(double x, double y) {
    return x + y;
}

int main() {
    double x0 = 0;
    double y0 = 1;
    double h = 0.05;
    double target_x = 0.20;

    while (x0 < target_x) {
        y0 = y0 + h * f(x0, y0);
        x0 += h;
    }

    double predicted_y = y0;
    
    double true_y = exp(0.20) - 0.20 - 1;
    
    double error = fabs(true_y - predicted_y);
    
    cout << "Predicted y(0.20) using Euler's method: " << predicted_y << endl;
    cout << "Error: " << error << endl;
    
    return 0;
}

File: bantuin/amin/1.py
# Definisikan fungsi PDB
def f(x, y):
    return x + y

# Inisialisasi nilai awal
x0 = 0
y0 = 1
h = 0.05
target_x = 0.20

# Perhitungan Metode Euler
while x0 < target_x:
    y0 = y0 + h * f(x0, y0)
    x0 += h

# Hasil perkiraan nilai y(0.20)
predicted_y = y0

# Solusi sejati
import math
true_y = math.exp(0.20) - 0.20 - 1

# Menghitung galat
error = abs(true_y - predicted_y)

print("Perkiraan y(0.20) menggunakan Metode Euler:", predicted_y)
print("Galat:", error)
